### DISCUSSION TITLE
Customizable horizon leveling

### DISCUSSION THREAD

**Discussion by kouyoumin** [gopro/labs#514] [Feature Requests]
In some use cases, complete 360-degree horizon lock is not necessary but only a small tilt correction needed.
Is it possible to set a customizable correction limit, say, 5°, to horizon leveling, and make FOV wider (than linear lens)?

I've tried max lens mod for both wide FOV and HL, but image quality is visibly worse.

**Comment by dnewman-gpsw** [comment]
There is no control to set the horizon levelling angle.  For the wisest FOV with Horizon levelling it is better to turn off in camera stabilization, then use the Wide lens and doing stabilization and lens correction it in ReelSteady within GoPro Player.  It is significantly wider.

**Reply by kouyoumin** [reply]
Thanks but stabilization cannot be disabled in timewarp mode.
(I need auto speed and speed ramp so no timelapse mode)

Adaptive cropping has been implemented in auto boost.
I think it's not difficult to apply similar technique (less crop when possible) to horizon leveling (and timewarp mode).

**Reply by dnewman-gpsw** [reply]
You didnt mention TW the original question.  The first part still applies "there is no control to set the horizon levelling angle." As the is nothing for Labs to hook on to, this will not be possible. However  using a Wide lens on TW is still more FOV then Linear, even with post correction.

**Reply by kouyoumin** [reply]
I tried GoPro Player but it says "Horizon leveling is not yet supported for Hypersmooth videos."

**Reply by dnewman-gpsw** [reply]
This might be job for HERO12 with Max Lens Mod 2.0, as it resolves 4K.  I can't think I another solution at the moment.

**Reply by kouyoumin** [reply]
My biggest trouble using HERO11 + Max Lens Mod is no speed ramp (switching to real speed) in max timewarp mode. No idea why.
Will HERO12 allow that?

**Reply by dnewman-gpsw** [reply]
It doesn't in my testing.  I can only speculate why, I too like the real-time speed control.

**Reply by kouyoumin** [reply]
Just got my HERO12. There's still no speed ramp in max timewarp. Sad.
